src/assets/scripts/resources-dynamic-handler.js

```diff
@@ -0,0 +1,145 @@
+		if (!isInitialRender) {
+			renderFilters(selectedTopics, selectedTypes);
+		}
+		if (filterQuery) {
+			renderSearchResults(resultsToDisplay.length, resourcesData.resourceTopics, resourcesData.resourceTypes);
+		}
+		renderResources(resultsToDisplay, resourcesData.resourceTopics, resourcesData.resourceTypes);
+		if (pagination) {
+			renderPagination(pagination);
+		}
+
+		// Clicking the expand button on the filter header opens/closes the filter
+		const expandButtons = document.querySelectorAll('.filter-expand-button');
+
+		for (let i = 0; i < expandButtons.length; i++) {
+			// Add event listener for expand buttons
+			expandButtons[i].addEventListener('click', (e) => {
+				e.preventDefault();
+				e.stopPropagation();
+				const currentExpandedValue = expandButtons[i].getAttribute('aria-expanded');
+				const expandedState = currentExpandedValue === 'true' ? 'false' : 'true';
+				const filterType = expandButtons[i].getAttribute('id').split('-')[1];
+				expandButtons[i].setAttribute('aria-expanded', expandedState);
+				expandButtons[i].setAttribute('aria-label', `${expandedState === 'true' ? "collapse" : "expand"} the ${filterType} filter`);
+				
+				// Store expanded status into local storage, so that expanded status for specific filter section is remembered.
+				localStorage.setItem(filterType, expandedState);
+
+				// Open/close the appropriate filter
+				// Find the filter body by using its position relative to the button as well as the css selector
+				// since there are two elements that match the selector (one each for static the and dynamic views).
+				// Clicking on one of expand buttons only opens the form that this button corresponds to.
+				const filterBodySelector = `#filterBody-${filterType}`;
+				const filter = $(expandButtons[i]).siblings(filterBodySelector);
+				filter[expandedState === 'false' ? 'hide' : 'show']();
+			});
+		}
+
+
+		// Clicking 'reset filter' button redirects the page to the initial state without search term and filtering conditions
+		if (filterQuery) {
+			document.querySelector('.reset-button').addEventListener('click', () => {
+				localStorage.setItem('setFocusOn', '.apply-button');
+				window.location = '/resources';
+			});
+		}
+
+		// Clicking 'apply filter' button redirects the page with applied filter options
+ 		document.querySelector('.apply-button').addEventListener('click', () => {
+			localStorage.setItem('setFocusOn', '.apply-button');
+		});
+
+		// Save element to focus after a filer option is removed by clicking on applied filter options
+		for (const filterTag of document.querySelectorAll('.filter-tag')) {
+			filterTag.addEventListener('click', (e) => {
+				const filter = $(filterTag);
+				if (filter.prev().length > 0) {
+					localStorage.setItem('setFocusOn', `#${filter.prev()[0].getAttribute('id')}`);
+				} else {
+					localStorage.setItem('setFocusOn', `#${filter.next()[0].getAttribute('id')}`);
+				}
+			});
+		}
+
+		restoreFocus();
```

Comment:
It might be better to move `restoreFocus()` to before `bindEventListeners()` since this is a part of actions to apply to the page while `bindEventListeners()` is static binding without performing any action.